### DISCUSSION TITLE
Using unocss as a vite plugin instead of unocss-cli

### DISCUSSION THREAD

**Discussion by TechAkayy** [unocss/unocss#2004] [Q&A]
Hi there, 

So, here I have a project which doesn't use any framework per-se

**Vite:** Basically, I'm have a main.js entry file and I use Vite in lib mode to build it in watch mode, and I use the compiled js file in my static html page (script tag in the body). 

`		"watch": "vite build --watch"` // my package.json script-1

**Unocss-cli:** Now, I'm also using unocss preset-icons in my static html page. And I'm using unocss-cli with glob patterns (unocss.config.ts) to my static html page. Unocss-cli beautifully picks only the icons I use and outputs uno.css, which in turn I use in my static html pages (link tag in the head).

`		"uno-watch": "unocss --watch",` // my package.json script-2

Both, vite's watch build & unocss-cli's watch build works perfectly. 

**Working Fine - the combo way!**
Nicely working repo - using Vite watch + Unocss-cli watch:
https://github.com/TechAkayy/pg-tailwind3-vite-app
https://stackblitz.com/github/TechAkayy/pg-tailwind3-vite-app?file=package.json

**Throwing an error - the vite way!**
It's just that I know I can use unocss as a vite plugin, and I'm wondering if it's possible to use Vite only and get rid of using unocss-cli? Can anyone help me on this. I tried moving config from unocss.config.js to vite.config.js, and use unocss only as a vite plugin but it's throwing some addWatchFile error like this: 

![image](https://user-images.githubusercontent.com/73772701/207889683-9e4fe7e8-9395-456f-a059-a3ca5fde1195.png)

Repo - I'm getting the above error when using vite watch only with unocss as a vite plugin:
https://github.com/TechAkayy/pg-tailwind3-vite-app-without-unocss-cli
https://stackblitz.com/github/TechAkayy/pg-tailwind3-vite-app-without-unocss-cli?file=package.json

In general, I just wanted to know if I'm missing something fundamentally in my second trial where I'm trying to use unocss as a vite plugin only. Even though I already have a working solution (vite-watch + unocss-cli-watch), simplifying to a single watch helps in the long run and in understanding the magic under the hood. Thanks in advance!

**Comment by TechAkayy** [comment]
Dear all, just checking to see if anyone could help :-) Cheers!

**Comment by sibbng** [comment]
I have a hacky solution for this, but I can not pinpoint the source of the problem. I need to know what causes this issue. Removing `tailwindcss` or `uno.css` fixes the issue. I tried re-creating the error with https://github.com/antfu/vitesse-lite but no luck.

**Reply by TechAkayy** [reply]
Oh woww.. thanks a ton for the hacky one ;-) I think I will probably raise this as a bug, clearly something seems to be wrong! Thanks again :-)
